### ISSUE TITLE
Create Build Container: PowerShell exited with code '1'

### ISSUE THREAD

**Issue by krregg** [microsoft/nav-docker#399]
I am not sure if it's ok to post this issue here, but anyway I will try.

I am following this case: [https://blogs.msdn.microsoft.com/freddyk/2018/11/12/developing-business-central-extensions-part-3/](https://blogs.msdn.microsoft.com/freddyk/2018/11/12/developing-business-central-extensions-part-3/)

I am on the step "Create a build pipeline". The build process exits with an error when trying to create Build Container. The exact error is: "/home/vsts/work/1/s/scripts/Create-Container.ps1 : The term 'New-NavContainer' is not recognized as the name of a cmdlet, function, script file, or operable program."

Before that is the step where NavContainerHelper is installed, all looks fine, I can retrieve the module version. It's strange that I get an error for cmdlet even if the NavContainerHelper was installed.

Here is a link to the print screen of my build steps: https://ibb.co/gMhByJF

Here is a link to print screen with an error when creating a build container: [https://ibb.co/QJ6xJ90](https://ibb.co/QJ6xJ90)

I noticed one thing with Agent Pools. For every build, I did there appeared a new agent as some ubuntu. I am not sure why this is happening. Here is a print screen: https://ibb.co/kQPcKYx

@freddydk Can you help me with this somehow? I would appreciate it a lot!

**Comment by freddydk** [comment]
On the first screen shot, it looks like it selects an Ubuntu 1604 build agent to run your script on. Have no idea why installing NavContainerHelper doesn't fail - but it certainly doesn't support Linux:-)
You need to use your own agent

**Comment by krregg** [comment]
I found it, I chose the wrong pool. But now I have another error and I have no clue...

```
Update-Module : Module 'navcontainerhelper' (installed at'C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper
\0.6.2.92') cannot be updated because Administrator rights are required to change that directory. Log on to the compute
r with an account that has Administrator rights, and then try again. You can also try running the Windows PowerShell se
ssion with elevated rights (Run as Administrator).
At C:\azagent\A1\_work\1\s\scripts\Install-NavContainerHelper.ps1:25 char:13
+             Update-Module -Name navcontainerhelper -Force -RequiredVe ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (navcontainerhelper:String) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : AdminPrivilegesAreRequiredForUpdate,Update-Module
```

What can I do in this case? How to set Admin Privileges?

**Comment by freddydk** [comment]
I think this is because your agent is not running as administrator.
(found here: https://stackoverflow.com/questions/36550343/run-powershell-or-command-prompt-as-administrator-during-visual-studio-team-serv)

**Comment by krregg** [comment]
I reinstalled the agent on my machine and now it works fine. But now I got something else...

```
Downloading C:\ProgramData\NavContainerHelper\Extensions\HiGregor-bld\my\license.flf
Download-File : Exception calling "DownloadFile" with "2" argument(s): "The underlying connection was closed: Could not
 establish trust relationship for the SSL/TLS secure channel."
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.2.93\ContainerHandling\New-NavContainer.ps1:793 ch
ar:9
```

This happens in create Build Container. I thought the problem is with the location of my license file when running New-NavContainer command, I fixed it, but no change. This is a container problem right?

**Comment by freddydk** [comment]
No, I don't think this is a container problem. This happens before it even starts to create the container.
It looks like the URL you provide for your licensefile is https with a self signed certificate, meaning that the build server doesn't trust the location.
I cannot ask you to provide the URL of your license file, but the above message is the direct error from trying to download the file.

**Comment by krregg** [comment]
Yes, I agree. I had to rest and look again. :)

I found where the problem was. The URL to my license file had problems with SSL. My mistake, I didn't set up a cert. on my server properly. Now it's working and build is successful. 

I wanted to learn this process asap but when you rush sometimes you don't see everything well. Now it's more clear to me how it's working after trying many times.

I just want to say I am really thankful for your kind support, @freddydk! I really appreciate it!

Thank you and have a nice day!
